File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {

       HashSet<Character > hs = new HashSet(); 
       int ans=0;
       for(Character ch : word.toCharArray()){
        if(Character.isLowerCase(ch)){
            hs.add(ch);
        }

       }

        for(Character ch : word.toCharArray()){
            if(Character.isUpperCase(ch)){
               if(hs.contains(Character.toLowerCase(ch))){
                ans++;
                hs.remove(Character.toLowerCase(ch));
               }
            }
        }
        return ans;
    }
}

File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        ls = list(word)
        pt1 = 0
        pt2 = len(ls)-1
        newls = []
        while(pt1<len(ls)-1):
            if ord(ls[pt1]) - ord(ls[pt2]) == 32 or ord(ls[pt1]) - ord(ls[pt2]) == -32 :
                newls.append(ls[pt1].lower())
            pt2-=1
            if pt2 == pt1:
                pt1+=1
                pt2 = len(ls)-1
        print(newls)
        newset = set(newls)
        return len(newset)